### PR TITLE
Search Poc: Add metrics

### PR DESCRIPTION
**Pull request by owensmallwood**
fixes https://github.com/grafana/grafana-org/issues/224

**Changes**
- can configure index path in grafana ini as:
```
[unified_storage]
index_path=/etc/foo/bar
```
- adds metrics: index size (bytes), time to build index (seconds), total docs indexed, latency for indexing new event (from event creation to finished indexing)

### CODE REVIEW

File: pkg/storage/unified/resource/index.go [hunk 1]
```diff
@@ -98,29 +107,37 @@ func (i *Index) Init(ctx context.Context) error {
...
+	err = shard.index.Index(res.Uid, res)
 	if err != nil {
 		return err
 	}
-	err = shard.index.Index(res.Metadata.Uid, indexableResource)
-	if err != nil {
-		return err
+
+	// record latency from when event was created to when it was indexed
+	latencySeconds := float64(time.Now().UnixMicro()-data.Value.ResourceVersion) / 1e6
```
Comment (scottlepp) [inline pkg/storage/unified/resource/index.go:136]: resourceversion is a timestamp?

File: pkg/storage/unified/resource/index_mapping.go [hunk 2]
```diff
@@ -25,6 +26,7 @@ type IndexedResource struct {
 }
 
 func (ir IndexedResource) FromSearchHit(hit *search.DocumentMatch) IndexedResource {
+	ir.Uid = hit.Fields["Uid"].(string)
 	ir.Kind = hit.Fields["Kind"].(string)
 	ir.Name = hit.Fields["Name"].(string)
 	ir.Namespace = hit.Fields["Namespace"].(string)
```
Comment (scottlepp) [inline pkg/storage/unified/resource/index_mapping.go:32]: might panic if the field doesn't exist or is nil.  not sure if that could ever happen

File: pkg/storage/unified/resource/index_server.go [hunk 3]
```diff
@@ -86,10 +88,21 @@ func (is *IndexServer) Init(ctx context.Context, rs *server) error {
 	return nil
 }
 
-func NewResourceIndexServer() ResourceIndexServer {
-	return &IndexServer{
-		log: slog.Default().With("logger", "index-server"),
+func NewResourceIndexServer(cfg *setting.Cfg) ResourceIndexServer {
+	// register metrics
```
Comment (scottlepp) [inline pkg/storage/unified/resource/index_server.go:92]: seems this comment is in the wrong place?

**Review by scottlepp — APPROVED** [review]

File: pkg/storage/unified/resource/index.go (hunk 1 above)
Comment (owensmallwood) [reply pkg/storage/unified/resource/index.go:136]: yeah, it is now bc of [this PR](https://github.com/grafana/grafana/pull/92638)

File: pkg/storage/unified/resource/index_mapping.go (hunk 2 above)
Comment (owensmallwood) [reply pkg/storage/unified/resource/index_mapping.go:32]: yeah, I think that's true for all these. But, they all _should_ exist afaik. Good enough for the PoC, but we should add some more graceful error handling for unpacking the search results at some point.

File: pkg/storage/unified/resource/index_server.go (hunk 3 above)
Comment (owensmallwood) [reply pkg/storage/unified/resource/index_server.go]: woops! removed 👍

**Pull request merged**